server/src/http_login.go

```diff
@@ -131,6 +132,35 @@ func httpLogin(c *gin.Context) {
 				http.Error(w, "That is not the correct password.", http.StatusUnauthorized)
 				return
 			}
+
+			// If they have provided a new password, replace the old one
```

Comment:
Is there a reason why this code should be inside the `else` from `if user.OldPasswordHash.Valid {`? It seems like it's indepedent of whether they had an old hash or not

Reply:
Not really, I can rewrite (and also make the password hashing an independent function)